lib/fluent/plugin/out_s3.rb

```diff
@@ -461,7 +461,8 @@ def check_s3_path_safety(conf)
         log.warn "The default value of s3_object_key_format will use ${chunk_id} instead of %{index} to avoid object conflict in v2"
       end
 
-      if (@buffer_config.flush_thread_count > 1) && ['${chunk_id}', '%{uuid_flush}'].none? { |key| @s3_object_key_format.include?(key) }
+      is_working_on_multi_thread = @buffer_config.flush_thread_count > 1 || system_config.workers > 1
+      if is_working_on_multi_thread && ['${chunk_id}', '%{uuid_flush}'].none? { |key| @s3_object_key_format.include?(key) }
         log.warn "No ${chunk_id} or %{uuid_flush} in s3_object_key_format with multiple flush threads. Recommend to set ${chunk_id} or %{uuid_flush} to avoid data lost by object conflict"
```

Comment:
I think we can tweak this error message to cover multiple workers,
so that it's clear to users why they got this warning.

In particular, I'm thinking something like this:

> "No ${chunk_id} or %{uuid_flush} in s3_object_key_format with multiple flush threads **(or multiple workers).**
> Recommend to set ${chunk_id} or %{uuid_flush} to avoid data lost by object conflict"

Reply:
Thank you!
I have fixed it.